### ISSUE TITLE
feat: Move CLI core logic to Typescript

### ISSUE THREAD

**Issue by ematipico** [webpack/webpack-cli#594]
**Is your feature request related to a problem? Please describe.**
As part the Typescript transition, I think the entire cli should be moved to Typescript. Maybe during the refactoring or straight after.

**Describe the solution you'd like**
As said, it can be done after the refactoring

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by ev1stensberg** [comment]
I don't think that we should, makes it harder for FT contributors to look at code. As apart of #245

**Comment by ematipico** [comment]
So why we started doing this at the first place?

The community has decided and we decided to move to typescript, hence we have to do it fully, no half ways. The "thing" of the code it's how we write the code, no matter the language.

If the code is commented, organized, well written, etc. The first contribution will come easily. Doesn't matter the language. Typescript has types and helps with errors.

I'm just saying that as part of this migration, we maintainers have to make fully, not halfway.
